mne/channels/channels.py

```diff
@@ -464,8 +466,34 @@ def set_montage(self, montage, set_dig=True, verbose=None):
         .. versionadded:: 0.9.0
         """
         from .montage import _set_montage
-        _set_montage(self.info, montage, update_ch_names=False,
-                     set_dig=set_dig)
+        from ..channels import Montage, DigMontage
+
+        _validate_type(montage, item_name='montage',
+                       types=(Montage, DigMontage, Digitization,
+                              str, type(None)))
+
+        if isinstance(montage, (Montage, DigMontage, str, type(None))):
+            _set_montage(self.info, montage, update_ch_names=False,
+                         set_dig=set_dig)
+        else:
+            self.info['dig'] = montage
+
+            eeg_montage = [x for x in montage
+                           if x['kind'] == FIFF.FIFFV_POINT_EEG]
+            # remove the first EEG which is the reference
+            ref_pos = eeg_montage[0]['r']
+            eeg_montage = eeg_montage[1:]
+
+            for ch, digpoint in zip(self.info['chs'], eeg_montage):
+                ch['kind'] = _human2fiff['eeg']
+                ch['unit'] = FIFF.FIFF_UNIT_V
+                ch['coil_type'] = FIFF.FIFFV_COIL_EEG
+                ch['coord_frame'] = digpoint['coord_frame']
+                ch['loc'] = np.zeros(12)
+                ch['loc'][:3] = digpoint['r']
+                ch['loc'][3:6] = ref_pos
+                ch['loc'][[7, 11]] = 1.  # not clear why
```

Comment:
i know for meg coils, idx 3-11 are the basis for neuromag space, i believe: https://github.com/mne-tools/mne-python/blob/master/mne/io/kit/kit.py#L704